src/main/java/de/tum/cit/aet/artemis/programming/service/localvc/LocalVCServletService.java

```diff
@@ -715,6 +771,125 @@ public static String getDefaultBranchOfRepository(Repository repository) {
+            UsernameAndPassword usernameAndPassword = extractUsernameAndPassword(authorizationHeader);
+            String userName = usernameAndPassword.username();
+            if (userName.equals(BUILD_USER_NAME)) {
+                return;
+            }
+            RepositoryActionType repositoryActionType = RepositoryActionType.PUSH;
+            var participation = getExerciseParticipationFromRequest(request);
+
+            vcsAccessLogService.ifPresent(service -> service.updateRepositoryActionType(participation, repositoryActionType));
+        }
+        catch (Exception ignored) {
+        }
+    }
+
+    /**
+     * Updates the VCS access log for clone and pull actions performed over SSH.
+     * <p>
+     * This method logs access information based on the SSH session and the root directory of the repository.
+     * It determines the repository action (clone or pull) based on the number of offers (`clientOffered`) and
+     * fetches participation details from the local VC repository URI.
+     *
+     * @param session       the {@link ServerSession} representing the SSH session.
+     * @param rootDir       the {@link Path} to the root directory of the repository.
+     * @param clientOffered the number of objects offered by the client in the operation, used to determine
+     *                          if the action is a clone (if 0) or a pull (if greater than 0).
+     */
+    @Async
+    public void updateVCSAccessLogForCloneAndPullSSH(ServerSession session, Path rootDir, int clientOffered) {
+        try {
+            if (session.getAttribute(SshConstants.USER_KEY).getName().equals(BUILD_USER_NAME)) {
+                return;
+            }
+            RepositoryActionType repositoryActionType = getRepositoryActionReadType(clientOffered);
+            var participation = retrieveParticipationFromLocalVCRepositoryUri(getLocalVCRepositoryUri(rootDir));
+            vcsAccessLogService.ifPresent(service -> service.updateRepositoryActionType(participation, repositoryActionType));
+        }
+        catch (Exception ignored) {
+        }
+    }
+
+    /**
+     * Adds a failed VCS access attempt to the log.
+     * <p>
+     * This method logs a failed clone attempt, associating it with the user and participation retrieved
+     * from the incoming HTTP request. It assumes that the failed attempt used password authentication.
+     *
+     * @param servletRequest the {@link HttpServletRequest} containing the HTTP request data.
+     */
+    public void createVCSAccessLogForFailedAuthenticationAttempt(HttpServletRequest servletRequest) {
+        try {
+            User user = getUserFromRequest(servletRequest);
+            var participation = getExerciseParticipationFromRequest(servletRequest);
+            vcsAccessLogService.ifPresent(service -> service.storeAccessLog(user, participation, RepositoryActionType.CLONE_FAIL, AuthenticationMechanism.PASSWORD, "", ""));
```

Comment:
Maybe add a note that this is missleading? or maybe rather in the client?

Reply:
I changed it, so it's clear whether a password or token was used :+1: